Add Test for .NET MGMT SDK Resources Test

HarveyLink opened the pull request.

HarveyLink committed:
Add test for .NET MGMT SDK Resources

erich-wang commented on sdk/resources/Azure.Management.Resources/tests/Azure.Management.Resources.Tests.csproj:
```diff
@@ -1,5 +1,21 @@
   <ItemGroup>
     <ProjectReference Include="..\src\Azure.Management.Resources.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <None Update="SessionRecords\**\*.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <None Update="ScenarioTests\*.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
```
please remove it

HarveyLink replied:
```
<ItemGroup>
    <None Update="ScenarioTests\*.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </None>
  </ItemGroup>
```
This block is need to copy the test required json template file.

erich-wang commented on sdk/resources/Azure.Management.Resources/tests/Azure.Management.Resources.Tests.csproj:
```diff
@@ -1,5 +1,21 @@
+  <PropertyGroup>
+    <NoWarn>$(NoWarn);CA1812;SA1633;SA1649</NoWarn>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" />
+  </ItemGroup>
```
we should not use Newtonsoft.Json, please remove it

HarveyLink replied:
Done

erich-wang commented on sdk/resources/Azure.Management.Resources/tests/Helpers/Utilities.cs:
```diff
@@ -0,0 +1,56 @@
+﻿using System;
```
License header should be added to each cs file

```
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
```

HarveyLink replied:
Done

erich-wang commented on sdk/resources/Azure.Management.Resources/tests/InMemoryTests/DeploymentTests.InMemory.cs:
```diff
@@ -0,0 +1,967 @@
+            options.Transport = transport;
+
+            return InstrumentClient(new ResourcesManagementClient(
+                "ddd5f1d5-499c-45ee-a5ff-024d36ade650",
```
we should use TestEnvironment.SubscriptionId instead of hard code

HarveyLink replied:
Done

erich-wang commented on sdk/resources/Azure.Management.Resources/tests/InMemoryTests/DeploymentTests.InMemory.cs:
```diff
@@ -0,0 +1,967 @@
+        public async Task DeploymentTestsCreateValidateMessage()
+        {
+            var mockResponse = new MockResponse((int)HttpStatusCode.Created);
+            var content = JObject.Parse(@"{
```
please try JsonDocument.Parse("").RootElement

HarveyLink replied:
Done

erich-wang commented on sdk/resources/Azure.Management.Resources/tests/InMemoryTests/DeploymentTests.InMemory.cs:
```diff
@@ -0,0 +1,967 @@
+              "\"value\": \"tianotest04\"" +
+             "}" +
+            "}";
+            JsonElement jsonTemplate = JsonSerializer.Deserialize<JsonElement>(templateString);
+            JsonElement jsonParameter = JsonSerializer.Deserialize<JsonElement>(parameterString);
+            object template = jsonTemplate.GetObject();
+            object parameter = jsonParameter.GetObject();
+
+            var mockTransport = new MockTransport(mockResponse);
+            var client = GetResourceManagementClient(mockTransport);
+
+            var parameters = new Deployment
+            (
+                new DeploymentProperties(DeploymentMode.Incremental)
+                {
+                    Template = template,
+                    Parameters = parameter
+                }
+             );
```
Please try following way
```
var parameterString = File.ReadAllText(xxx);
JsonElement jsonParameter = JsonSerializer.Deserialize<JsonElement>(parameterString);
if(!jsonParameter.TryGetProperty("parameters", out JsonElement parameter))
{
    parameter = jsonParameter;
}
var parameters = new Deployment
(
    new DeploymentProperties(DeploymentMode.Incremental)
    {
        Template = File.ReadAllText(xxx);,
        Parameters = parameter.ToString()
    }
);
```

blueww committed:
[Storage] Add test case for accesstier (#12340)

yuzhangyi committed:
[Datafactory] Enable Excel SDK (#12282)

* Enable Excel SDK

* update

Co-authored-by: zhanyu2014 <zhanyu@microsoft.com>

christothes committed:
update Azure.ClientSdk.Analyzers to 20200529.1 (#12370)

maririos committed:
make it required (#12354)

christothes committed:
Tables: Query support for dictionary entities and CreateFilter (#12366)

* Support for dictionary entities and CreateFilter

* add comment to CreateFilter

* take the new Azure.ClientSdk.Analyzers and remove pragmas

* remove redundant .ToList() calls in tests

* make CreateFilter less prominant on the client

azabbasi committed:
Initial commit to master for Azure Digital Twins .Net SDK. (#12398)

jafreck committed:
Update Batch SDK regeneration instructions (#12347)

NarayanThiru committed:
K8sConfiguration .Net SDK - Create (#12227)

* K8sConfiguration .Net SDK - Create

* include k8s in CI

* Project version and dependencies update

Co-authored-by: nanthi <nanthi@NANTHI01>
Co-authored-by: Yeming Liu <Yeming.Liu@microsoft.com>

yangyuan committed:
Support Sku capabilities. Bump to 7.3.0 (#12328)

* Support Sku capabilities. Bump to 7.3.0

* Generate from AutoRest

jennifermarsman committed:
Removed color and imageType from the VisualFeature enum (#12375)

* Removed color and imageType from the VisualFeature enum

* Modified test data and removed changes to generated code (so the deprecated enum still exists in there)

* Regenerated CreateSkillsetReturnsCorrectDefinitionImageAnalysisKeyPhrase.json on dogfood

juniwang committed:
Generate new .NET SDK based on #9377 (#12364)

* Generate new .NET SDK based on #9377

* Update package version

HarveyLink committed:
Add fix for  Resources Tests

jsquire committed:
[Event Hubs Client] Migration Guide Updates (#12402)

The focus of these changes is to enhance the migration guide with additional
information about the client hierarchy and discuss the limitations around
processor types consuming legacy checkpoint data.

kinelski committed:
[FormRecognizer] Remove Operation type constructors (#12419)

kasobol-msft committed:
Release prep (#12422)

* Prepare Storage for release

* pr feedback.

* PR feedback.

* pr feedback.

* Release preparation

chidozieononiwu committed:
Update GitIgnore (#12368)

pakrym committed:
Enable return type and banned assembly analyzers (#12418)

azure-sdk committed:
Increment version for storage releases (#12423)

* Increment package version after release of Azure.Storage.Common

* Increment package version after release of Azure.Storage.Files.Shares

* Increment package version after release of Azure.Storage.Blobs

* Increment package version after release of Azure.Storage.Files.DataLake

* Increment package version after release of Azure.Storage.Queues

azure-sdk committed:
Increment version for storage releases (#12425)

* Increment package version after release of Azure.Storage.Common

* Increment package version after release of Azure.Storage.Files.Shares

* Increment package version after release of Azure.Storage.Blobs

* Increment package version after release of Azure.Storage.Files.DataLake

* Increment package version after release of Azure.Storage.Queues

Commit:
feat(Samples): Add Telemetry Sample (#12407)

feat(Samples): Add Telemetry Sample

christothes committed:
add some unit tests for expression parsing (#12421)

* add some unit tests for expression parsing

* refactor for TestCaseSource

* fixup naming

Commit:
samples(adt): Update ctor snippets, and update sample options (#12409)

vinagesh committed:
fix(e2e): Fix replay tests for differnt OS(s) (#12424)

Commit:
fix(adt): fix BasicDigitalTwin and use in an example (#12428)

azabbasi committed:
Update root readme file (#12430)

Commit:
samples(adt): add a sample for using a custom twin type for serialization (#12432)

isra-fel committed:
Update swagger_to_sdk_config.json (#12412)

@microsoft.azure/autorest.csharp@2.3.84 is a preview version. Going back to 82

scbedd committed:
Fix Incomplete Doc Upload (#12442)

* ensure that child directory items are copied

* bump template version so it can actually pass through the version checks + integration feed

azure-sdk committed:
Update AutoRest C# version to https://github.com/Azure/autorest.csharp/releases/download/3.0.0-dev.20200602.1/autorest-csharp-v3-3.0.0-dev.20200602.1.tgz (#12435)

pakrym committed:
Fix json serialization of geometries (#12440)

Commit:
samples(adt): add snippets for get and deserialize digital twin (#12441)

vinagesh committed:
fix(adt): Updated comments to generted more meaningful API documentation (#12443)

amnguye committed:
Escaping % (percent sign) segment in Uri.AppendToPath  (#12369)

* Escaping % percent sign in segment in AppendToPath

Commit:
webdocs(adt): test seealso, and adding links for more resources (#12471)

heaths committed:
Make collection properties read-only (#12459)

* Make collection properties read-only

Fixes #11712. Removes the setters per .NET guidelines.

* Resolve PR feedback

vinagesh committed:
feat(adt): Adding see-also section for sync methods (#12470)

Commit:
comments(adt): removing inaccurate comment (#12474)

heaths committed:
Remove FieldBuilder (#12464)

* Remove FieldBuilder

Resolves #12460

* Update public APIs

vinagesh committed:
feat(test): Adding tests to ensure that the test mode values are not changed (#12481)

heaths committed:
Ignore Azure.Management.KeyVault.Tests (#12483)

Works around #12211 by ignoring tests that fail sporadically for both
live and playback tests.

pakrym committed:
Default to more Newtonsoft-like response encoding in TestFramework (#12482)

Commit:
samples(adt): misc minor clean-up from Azure SDK review (#12488)

vinagesh committed:
feat(adt): Add exception tag to comments (#12475)

JoshLove-msft committed:
Add SessionReceiverOptions/SessionProcessorOptions (#12383)

pakrym committed:
Changelogs and version updates for June release (#12507)

Commit:
test(PublishTelemetry): Add E2E test for Publish Telemetry API (#12479)

azure-sdk committed:
Increment version for core releases (#12516)

pakrym committed:
Update versions of Azure.Core and Azure.Core.Experimental (#12520)

azure-sdk committed:
Update AutoRest C# version (#12525)

Co-authored-by: Pavel Krymets <pavel@krymets.com>

kasobol-msft committed:
Release Prep (#12526)

* Prepare Storage for release

* pr feedback.

* PR feedback.

* pr feedback.

* release prep

abhipsaMisra committed:
samples(adt): Update component property to be an integer (#12515)

* samples(adt): Update component property to be an integer

* samples(adt): Update dt property to be an integer

Co-authored-by: Abhipsa Misra <abhipsa.misra@microsoft.com>

azure-sdk committed:
Increment version for storage releases (#12529)

* Increment package version after release of Azure.Storage.Queues

* Increment package version after release of Azure.Storage.Common

* Increment package version after release of Azure.Storage.Files.Shares

* Increment package version after release of Azure.Storage.Blobs

* Increment package version after release of Azure.Storage.Files.DataLake

Commit:
samples(adt): more misc changes per Azure SDK review (#12532)

Commit:
feat(samples): Add an introduction section to the samples page (#12524)

HarveyLink committed:
Merge branch 'v-minghc/resources-test'
